athenahealth/httpclient.go

```diff
@@ -232,19 +234,31 @@ func (h *HTTPClient) request(ctx context.Context, method, path string, body io.R
 		req.Header = headers
 	}
 
+	xRequestID := uuid.NewString()
+
 	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 	req.Header.Add("User-Agent", userAgent)
+	req.Header.Set("X-Request-Id", xRequestID)
+
+	if span, ok := tracer.SpanFromContext(ctx); ok {
```

Comment:
This span, if it exists, will not be the span that is used to trace the http http request. That span will be created by our wrapped http transport which occurs inside the `Do(req)` call. A better option is to _not_ make this library datadog aware and instead support optional configuration when we wrap the transport.
We can export a const to the header key and use that to extract the header and inject into the span in the caller.

Reply:
https://github.com/eleanorhealth/go-common/pull/29